src/components/TodoItem.tsx

```diff
@@ -0,0 +1,130 @@
+import classNames from 'classnames';
+import { useState } from 'react';
+import { Todo } from '../types/Todo';
+import { patchTodo } from '../api/todos';
+
+type Props = {
+  todo: Todo,
+  onDelete: (todoId: number) => void,
+  updateTodo: (updatedTodo: Todo) => void,
+  removeLoadingTodo: (todoId: number) => void,
+  addLoadingTodo: (todoId: number) => void,
+  loadingTodos: number[],
+  deletingTodoId: number | null,
+  isLoading: boolean,
+  updateTodoOnServer: (todoId: number, data: Partial<Todo>) => Promise<void>,
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  onDelete,
+  updateTodo,
+  removeLoadingTodo,
+  addLoadingTodo,
+  loadingTodos,
+  deletingTodoId,
+  isLoading,
+  updateTodoOnServer,
+}) => {
+  const { id, title, completed } = todo;
+
+  const [checked, setChecked] = useState(completed);
```

Comment:
this state duplicates the prop. It's a bad practice. We should use only one source of truth